flutter/lib/src/feedback/sentry_feedback_widget.dart

```diff
@@ -0,0 +1,266 @@
+
+class SentryFeedbackWidget extends StatefulWidget {
+  const SentryFeedbackWidget({
+    super.key,
+    this.associatedEventId,
+    this.hub,
+    this.title = 'Report a Bug',
+    this.nameLabel = 'Name',
+    this.namePlaceholder = 'Your Name',
+    this.emailLabel = 'Email',
+    this.emailPlaceholder = 'your.email@example.org',
+    this.messageLabel = 'Description',
+    this.messagePlaceholder = 'What\'s the bug? What did you expect?',
+    this.submitButtonLabel = 'Send Bug Report',
+    this.cancelButtonLabel = 'Cancel',
+    this.isRequiredLabel = '(required)',
+    this.isNameRequired = false,
+    this.isEmailRequired = false,
+  }) : assert(associatedEventId != const SentryId.empty());
+
+  final SentryId? associatedEventId;
+  final Hub? hub;
+
+  final String title;
+
+  final String nameLabel;
+  final String namePlaceholder;
+  final String emailLabel;
+  final String emailPlaceholder;
+  final String messageLabel;
+  final String messagePlaceholder;
+
+  final String submitButtonLabel;
+  final String cancelButtonLabel;
+
+  final String isRequiredLabel;
+
+  final bool isNameRequired;
+  final bool isEmailRequired;
+
+  @override
+  _SentryFeedbackWidgetState createState() => _SentryFeedbackWidgetState();
+}
+
+class _SentryFeedbackWidgetState extends State<SentryFeedbackWidget> {
+  final TextEditingController _nameController = TextEditingController();
+  final TextEditingController _emailController = TextEditingController();
+  final TextEditingController _messageController = TextEditingController();
+
+  String? _name;
+  String? _email;
+  String? _message;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(widget.title),
+      ),
+      body: Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Column(
+          children: [
+            Expanded(
+              child: SingleChildScrollView(
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  mainAxisSize: MainAxisSize.min,
+                  children: [
+                    Row(
+                      children: [
+                        Text(
+                          key: const ValueKey('sentry_feedback_name_label'),
+                          widget.nameLabel,
+                          style: Theme.of(context).textTheme.labelMedium,
+                        ),
+                        const SizedBox(width: 4),
+                        if (widget.isNameRequired)
+                          Text(
+                            key: const ValueKey(
+                                'sentry_feedback_name_required_label'),
+                            widget.isRequiredLabel,
+                            style: Theme.of(context).textTheme.labelMedium,
+                          ),
+                      ],
+                    ),
+                    const SizedBox(height: 4),
+                    TextField(
```

Comment:
I would advise to use the [`Form`](https://api.flutter.dev/flutter/widgets/Form-class.html) and [`TextFormField`](https://api.flutter.dev/flutter/material/TextFormField-class.html) here, since they have validation built-in.

Reply:
will check this out, thx

Reply:
neat, thx for the suggestion!

Reply:
You're welcome